### PR TITLE
chore: verify that SAs are applied only once

### PR DESCRIPTION
**Pull request by MatousJobanek**
paired PR: https://github.com/codeready-toolchain/member-operator/pull/366

### CODE REVIEW

File: test/e2e/parallel/serviceaccount_test.go [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+	require.NoError(t, err)
+	assert.Equal(t, "stay", sa.Annotations["should"])
+
+	// verify the secrets created for SA
+	// list all secrets in the namespace
+	secrets := &corev1.SecretList{}
+	require.NoError(t, member.Client.List(context.TODO(), secrets, client.InNamespace(mur.Name)))
+	for _, secret := range secrets.Items {
+		// filter only those that are marked to be owned by the SA
+		if secret.Annotations[corev1.ServiceAccountNameKey] == sa.Name {
```
Comment (xcoulon) [inline test/e2e/parallel/serviceaccount_test.go:66]: you could probably count the number of secrets with the `corev1.ServiceAccountNameKey` annotation matching the `sa.Name` and compare with the number of secrets in the SA itself (`sa.Secrets`). 
Ah, but you would not have the name of the secret in the assertion failure message, then...

File: test/e2e/parallel/serviceaccount_test.go [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+	sa, err := member.WaitForServiceAccount(mur.Name, fmt.Sprintf("appstudio-%s", mur.Name))
+	require.NoError(t, err)
+
+	// when we add an annotation to the SA resource then it should stay there
+	sa.Annotations = map[string]string{
+		"should": "stay",
+	}
+	require.NoError(t, member.Client.Update(context.TODO(), sa))
+
+	// drop the SpaceRoles annotation from the namespace to trigger the reconciliation
```
[the remark this answers is not in the document]
Comment (MatousJobanek) [reply test/e2e/parallel/serviceaccount_test.go:50]: could work too, but this is easier

File: test/e2e/parallel/serviceaccount_test.go (hunk 1 above)
Comment (MatousJobanek) [reply test/e2e/parallel/serviceaccount_test.go]: updated the test to make it more robust

**Pull request merged**